src/optimal_image/lib.rs

```diff
@@ -1,6 +1,60 @@
-extern crate image;
 extern crate dssim;
+extern crate image;
 extern crate imgref;
 extern crate lodepng;
 
+use std::cmp::Ordering;
+
+#[derive(Debug, Clone)]
+pub enum ImageFormat {
+    JPEG,
+    PNG,
+    WEBP,
+    JPEGXR,
+}
+
+#[derive(Debug, Clone)]
+pub struct EncodingConfig {
+    format: ImageFormat,
+    quality: u8,
```

Comment:
What does `quality` denote? Can we document using doc comments? In general, lets try to emphasize on documentation to be done well for users. Current state of RUST library docs are less useful for beginners as compared to go docs